iris-mpc-cpu/src/network/value.rs

```diff
@@ -16,20 +17,231 @@ pub enum NetworkValue {
 }
 
 impl NetworkValue {
+    fn get_descriptor_byte(&self) -> u8 {
+        match self {
+            NetworkValue::PrfKey(_) => 0x01,
+            NetworkValue::RingElementBit(bit) => {
+                if bit.convert().convert() {
+                    0x12
+                } else {
+                    0x02
+                }
+            }
+            NetworkValue::RingElement16(_) => 0x03,
+            NetworkValue::RingElement32(_) => 0x04,
+            NetworkValue::RingElement64(_) => 0x05,
+            NetworkValue::VecRing16(_) => 0x06,
+            NetworkValue::VecRing32(_) => 0x07,
+            NetworkValue::VecRing64(_) => 0x08,
+        }
+    }
+
     pub fn to_network(&self) -> Vec<u8> {
-        bitcode::serialize(self).unwrap()
+        let mut res = vec![];
```

Comment:
can probably get rid of a few allocations for better perf:
* here, we probably want to have already a vec with capacity 1+ self.byte_len (need a function for this), otherwise this will allocate multiple times (at least 2 times for many variants, multiple times for Vec variants)